lib/rules/no-loss-of-precision.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * @fileoverview Rule to flag numbers that will lose significant figure precision at runtime
+ * @author Jacob Moore
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        type: "problem",
+
+        docs: {
+            description: "disallow literal numbers that lose precision",
+            category: "Possible Errors",
+            recommended: true,
+            url: "https://eslint.org/docs/rules/no-loss-of-precision"
+        },
+        schema: [],
+        messages: {
+            noLossOfPrecision: "This number will lose precision when stored as a Number type."
+        }
+    },
+
+    create(context) {
+
+        /**
+         * Returns whether the node is number literal
+         * @param {Node} node the node literal being evaluated
+         * @returns {boolean} true if the node is a number literal
+         */
+        function isNumber(node) {
+            return typeof node.value === "number";
+        }
+
+        /**
+         * Checks whether the number is not base ten
+         * @param {ASTNode} node the node being evaluated
+         * @returns {boolean} true if the node is not in base ten
+         */
+        function isNotBaseTen(node) {
+            const prefixes = ["0x", "0X", "0b", "0B", "0o", "0O"];
+
+            return prefixes.some(prefix => node.raw.startsWith(prefix)) ||
+            (node.raw.startsWith("0") && !node.raw.startsWith("0."));
```

Comment:
I think this function is incorrect on the following literals:

* `0e5` (base 10 scientific notation)
* `0129` (base 10 because it contains a non-octal digit)